app/connectors/PSRConnector.scala

```diff
@@ -45,22 +44,18 @@ class PSRConnector @Inject()(appConfig: FrontendAppConfig, http: HttpClient) {
     s"$baseUrl/pension-scheme-return/psr/versions/$pstr?startDate=$startDate"
 
   def submitPsrDetails(
-    psrSubmission: PsrSubmission,
-    optFallbackCall: Option[Call] = None
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Unit] =
+    psrSubmission: PsrSubmission
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[String, Boolean]] =
     http
       .POST[PsrSubmission, HttpResponse](
         submitStandardUrl,
         psrSubmission
       )
       .map { response =>
         response.status match {
-          case NO_CONTENT => NoContent
+          case NO_CONTENT => Right(true)
           case _ =>
-            val url = optFallbackCall.fold(appConfig.urls.managePensionsSchemes.dashboard) { fallbackCall =>
-              fallbackCall.url
-            }
-            throw PostPsrException(s"{${response.status}, ${response.json}}", url)
+            Left(s"{${response.status}, ${response.json}}")
```

Comment:
Just an idea, would it be better if we return Future[Either[String, Unit]] ? Because that boolean doesn't make sense to me.

```suggestion
              def submitPsrDetails(
    psrSubmission: PsrSubmission
  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[String, Unit]] =
    http
      .POST[PsrSubmission, HttpResponse](
        submitStandardUrl,
        psrSubmission
      )
      .map { response =>
        response.status match {
          case NO_CONTENT => Right(())
          case _ =>
            Left(s"{${response.status}, ${response.json}}")
        }
      }
```

```
            case Right(()) =>
              auditService.sendExtendedEvent(buildAuditEvent(taxYear, loggedInUserNameOrBlank(request)))
              Future.unit
```

Reply:
Yes, that makes sense, I'm going to change that.

Reply:
I changed this.